api/src/database/models/asset.model.ts

```diff
@@ -14,11 +14,11 @@ class Asset extends Model<
   declare id: CreationOptional<number>;
   declare assetTag: string;
   declare name: string;
-  declare serialNumber: CreationOptional<string>;
-  declare modelNumber: CreationOptional<string>;
-  declare nextAuditDate: CreationOptional<Date>;
-  declare createdAt: CreationOptional<Date>;
-  declare updatedAt: CreationOptional<Date>;
+  declare serialNumber: CreationOptional<string | null>;
```

Comment:
CreationOptional automatically defaults to Null but happy to have it referenced.

Reply:
Looking at the type definition for `CreationOptional`, it is `T extends null | undefined ? T : T & { [CreationAttributeBrand]?: true }`. So it does not auto-default to `null`. If you remove the explicit `| null`, then the field can be assigned to `T`, whereas, with the explicit `| null` it cannot - instead requiring `T | null`, enforcing the types to reflect the values.